ui/api.py

```diff
@@ -0,0 +1,67 @@
+"""
+API methods
+"""
+
+from django.conf import settings
+
+from ui.encodings import EncodingNames
+from ui.util import get_et_preset, get_bucket, get_et_job
+from ui.models import VideoFile, VideoThumbnail
+
+
+def process_transcode_results(video, job):
+    """
+    Create VideoFile and VideoThumbnail objects for a Video based on AWS ET job output
+
+    Args:
+        video(Video): Video object to which files and thumbnails belong
+        job(JSON): JSON representation of AWS ET job output
+    """
+
+    for playlist in job['Playlists']:
+        VideoFile.objects.get_or_create(
+            s3_object_key='{}.m3u8'.format(playlist['Name']),  # This assumes HLS encoding
+            defaults={
+                'video': video,
+                'bucket_name': settings.VIDEO_S3_TRANSCODE_BUCKET,
+                'encoding': EncodingNames.HLS,
+                'preset_id': ','.join([output['PresetId'] for output in job['Outputs']]),
+            }
+
+        )
+    for output in job['Outputs']:
+        if 'ThumbnailPattern' not in output:
+            continue
+        thumbnail_pattern = output['ThumbnailPattern'].replace("{count}", "")
+        preset = get_et_preset(output['PresetId'])
+        bucket = get_bucket(settings.VIDEO_S3_THUMBNAIL_BUCKET)
+        for thumb in bucket.objects.filter(Prefix=thumbnail_pattern):
+            VideoThumbnail.objects.get_or_create(
+                s3_object_key=thumb.key,
+                defaults={
+                    'video': video,
+                    'bucket_name': settings.VIDEO_S3_THUMBNAIL_BUCKET,
+                    'preset_id': output['PresetId'],
+                    'max_height': int(preset['Thumbnails']['MaxHeight']),
+                    'max_width': int(preset['Thumbnails']['MaxWidth'])
+                }
+            )
+
+
+def refresh_status(video):
+    """
+    Check the encode job status & if not complete, update the status via a query to AWS.
+
+    Returns:
+        str: The video's latest encode job status
+
+    """
+    if video.status in ['Complete', 'Error']:
```

Comment:
i think you can just remove all the return code and just execute lines 62-66 if the status isn't complete or error. See my other comment in the view.